### PR TITLE
macos CI using github actions

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
Here's a run on my fork with free Github-hosted actions runners. https://github.com/isuruf/pocl/actions/runs/5203572952

**Comment by pjaaskel** [comment]
Thanks! The main question before pulling this is in: Do you now volunteer to become the "MacOS maintainer" who will react to issues found by that CI? The current lead developers of PoCL do not use Macs and thus can't fix issues nor are able to spend time on them.

**Comment by isuruf** [comment]
Sure. Please ping me if you run across any issues.

**Comment by pjaaskel** [comment]
Can you rebase this also on release_4_0?

**Comment by isuruf** [comment]
Done

File: .github/workflows/build_cmake_macos.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        # devel = with devel options
+        # cts_spirv = cts with SPIR-V
+        # tier1 = includes CTS without SPIR-V
+        # asan, tsan, ubsan = sanitizers
+        config: [basic]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: conda-incubator/setup-miniconda@v2
```
Comment (isuruf) [inline .github/workflows/build_cmake_macos.yml:41]: This needs to be added as an allowed action. It's used for installing conda to install llvm, clang, etc

Comment (franz) [reply .github/workflows/build_cmake_macos.yml:41]: added

**Pull request merged**